python/param_io.cpp

```diff
@@ -0,0 +1,114 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <fstream>
+#include <iomanip>
+
+#include <arbor/cable_cell.hpp>
+#include <arbor/morph/primitives.hpp>
+
+#include <arborio/jsonio.hpp>
+
+#include "error.hpp"
+#include "strprintf.hpp"
+
+namespace pyarb {
+void register_param_loader(pybind11::module& m) {
+    m.def("load_default_parameters",
+          [](std::string fname) {
+              std::ifstream fid{fname};
+              if (!fid.good()) {
+                  throw pyarb_error("Can't open file '{}'" + fname);
+              }
+
+              arb::cable_cell_parameter_set params;
+              try {
+                  params = arborio::load_cable_cell_parameter_set(fid);
+              }
+              catch (std::exception& e) {
+                  throw pyarb_error("Error loading default parameters from \"" + fname + "\": " + std::string(e.what()));
+              }
+
+              arb::cable_cell_global_properties G;
+              G.default_parameters = params;
+              try {
+                  arb::check_global_properties(G);
+              }
+              catch (std::exception& e) {
+                  throw pyarb_error("Error loading default parameters from \"" + fname + "\": default parameter check failed : " + std::string(e.what()));
+              }
+              return params;
+          },
+          "Load default model parameters from file.");
+
+    m.def("load_decor",
```

Comment:
All of these methods are either reading or writing JSON?

If so, then at least the name should reflect that.

Reply:
It is a pity that the Python JSON representation and our C++ nlohmann are not the same: then we could pass JSON objects instead of file names. We could then just construct the objects from json representations.

Reply:
I'll fix the names. 
As for the second point, do you mean doing something like: 
```
import json
with open(sys.argv[1]) as json_file:
    arbor.load_json_decor(json.load(json_file))
```
I didn't actually try it, but it seems a bit verbose.

Reply:
It is a bit more verbose than just passing in a file name.
But it is more "typey", because you don't need the `json` in the function name.
`load_decor` would be enough, and could take multiple different types.

It won't work as is, though, because we would have to teach our parser to understand Pythons JSON representation.

Reply:
I see, I agree that that would be nice to have.